scanner/cmd/scanner/main.go

```diff
@@ -221,7 +227,50 @@ func getServices(b *Backends) []grpc.APIService {
 	return srvs
 }
 
-// Close closes all the backends used by the scanner.
+// HealthCheck returns true if all configured backends are healthy and ready.
+func (b *Backends) HealthCheck(ctx context.Context) bool {
+	var checkList []func(context.Context) error
+	if b.Indexer != nil {
+		checkList = append(checkList, b.Indexer.Ready)
+	}
+	if b.RemoteIndexer != nil {
+		checkList = append(checkList, b.RemoteIndexer.Ready)
+	}
+	if b.Matcher != nil {
+		checkList = append(checkList, b.Matcher.Ready)
+	}
+	for _, check := range checkList {
+		if err := check(ctx); err != nil {
+			zlog.Error(ctx).Err(err).Msg("scanner is not ready")
+			return false
+		}
+	}
+	return true
+}
+
+// HealthRoutes returns HTTP routes for health checking the configured backends.
+func (b *Backends) HealthRoutes() (r []routes.CustomRoute) {
+	for n, h := range map[string]http.HandlerFunc{
+		"/health/liveness": func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Kubernetes does `/livez` and `/readyz`. I have seen the `z` paradigm before, but I can't say I know why it exists. Any thoughts about using that instead? Clair seems to use `/healthz` and `/readyz`

Reply:
I don't know either. My guess is it's a shortening convention that doesn't make much of a difference because we use the full name, similar to appending `ctl` to command-line tooling.